tests/lib/modem_trace/CMakeLists.txt

```
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(modem_trace_test)
+
+# create mock
+cmock_handle(${ZEPHYR_BASE}/../modules/hal/nordic/nrfx/drivers/include/nrfx_uarte.h )
+cmock_handle(${ZEPHYR_BASE}/../modules/debug/segger/SEGGER/SEGGER_RTT.h )
+
+# generate runner for the test
+test_runner_generate(src/modem_trace_test.c)
+
+target_include_directories(app PRIVATE src)
+
+# add test file
+target_sources(app PRIVATE src/modem_trace_test.c)
+
+# add unit under test
+target_sources(app PRIVATE ../../../lib/nrf_modem_lib/modem_trace.c)
+
+# include paths
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/../modules/hal/nordic/nrfx)
```

```suggestion
target_include_directories(app PRIVATE ${ZEPHYR_HAL_NORDIC_MODULE_DIR}/nordic/nrfx)
```
and similar on the lines below.

Remember to adjust according to the module referenced.